### PR TITLE
Datagrid: Hide from panel list if toggle disabled

### PR DESCRIPTION
**Pull request by mdvictor**
**What is this feature?**

Hides the datagrid panel if the ff is disabled

**Why do we need this feature?**

Datagrid will be released as experimental and will need to be entirely hidden if ff is disabled.

**Who is this feature for?**

Everyone

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/api/frontendsettings.go [hunk 1]
```diff
@@ -63,6 +64,10 @@ func (hs *HTTPServer) getFrontendSettings(c *contextmodel.ReqContext) (*dtos.Fro
 			continue
 		}
 
+		if panel.ID == "datagrid" && !hs.Features.IsEnabled(featuremgmt.FlagEnableDatagridEditing) {
```
Comment (baldm0mma) [inline pkg/api/frontendsettings.go:67]: Nitpick: I like to be explicit with my booleans just to make it super clear to anyone reading it, and to the IsEnabled function; I'd !! it. -> `!hs.Features.IsEnabled(!!featuremgmt.FlagEnableDatagridEditing)`

**Review by baldm0mma — APPROVED** [review]
Dope.

**Comment by supilee** [comment]
Datagrid is not visible anymore as expected BUT the panels that used datagrid panels as a source are still showing the data. Is that as designed?

**Comment by mdvictor** [comment]
> Datagrid is not visible anymore as expected BUT the panels that used datagrid panels as a source are still showing the data. Is that as designed?

I don't think that's a problem. The data remains even if it's a different panel (e.g: a table with data and other panels use that table as ds, if you then delete the table the other panels still have that data until you modify the query editor)

File: pkg/api/frontendsettings.go (hunk 1 above)
Comment (zoltanbedi) [reply pkg/api/frontendsettings.go:67]: Where are you seeing that the `FlagEnableDatagridEditing` can be undefined? Go doesn't even have undefined.

**Review by zoltanbedi — APPROVED** [review]

**Review by supilee — APPROVED** [review]

**Pull request merged**